activestorage/lib/active_storage/engine.rb

```diff
@@ -97,10 +98,22 @@ class Engine < Rails::Engine # :nodoc:
         ActiveStorage.content_types_allowed_inline = app.config.active_storage.content_types_allowed_inline || []
         ActiveStorage.binary_content_type = app.config.active_storage.binary_content_type || "application/octet-stream"
         ActiveStorage.video_preview_arguments = app.config.active_storage.video_preview_arguments || "-y -vframes 1 -f image2"
+        ActiveStorage.transformations_by_format = app.config.active_storage.transformations_by_format || {}
 
         ActiveStorage.replace_on_assign_to_many = app.config.active_storage.replace_on_assign_to_many || false
         ActiveStorage.track_variants = app.config.active_storage.track_variants || false
       end
+
+      config.after_initialize do |_|
+        invalid_formats = ActiveStorage.transformations_by_format.keys - ActiveStorage.web_image_content_types.map { |content_type| content_type.split("/")[1].to_sym }
+
+        if invalid_formats.present?
+          raise ArgumentError,
+                "You have defined transformations for formats that Active Storage does not consider web images (#{invalid_formats}) and will automatically convert to PNG. " \
+                "If you really wish to display images in these formats, add them to the 'web_image_content_types' config. " \
+                "Otherwise, remove them from 'transformations_by_format' and define transformations for the PNG format instead."
```

Comment:
@kaspth I've looked around and couldn't find a good example of the correct way to `raise` on bad configurations. I've added another `after_initialize` the the `configs` initializer and wrote the check here. Is this what you expected?

Also, english is not my first language, so I'd welcome some grammar improvements in this message. I've tried to make clear what is wrong and how to fix the problem. And in the end it also serves as the documentation I mentioned.